### PR TITLE
Fix rojo not found for sourcemaps on macOS

### PR DESCRIPTION
**Pull request by filiptibell**
Spawning the child process for creating the Rojo sourcemap now has a fallback where the shell configured in vscode settings is spawned interactively and the `rojo sourcemap` command runs in there. This means that the shell is responsible for loading the environment and `PATH` variable instead of vscode / the daemon that starts up GUI apps for macOS.

### CODE REVIEW

File: editors/code/src/spawn.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+const spawnNaive = (
+  command: string,
+  args: string[],
+  options?: SpawnOptionsWithoutStdio
+): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    // Spawn the wanted command directly
+    const cp = spawn(command, args, {
+      cwd: process.cwd(),
```
Comment (JohnnyMorganz) [inline editors/code/src/spawn.ts:70]: cwd should use the one passed in options if available
```suggestion
      cwd: options?.cwd ?? process.cwd(),
```

File: editors/code/src/spawn.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  return new Promise((resolve, reject) => {
+    // Spawn the configured vscode shell as if it is a
+    // normal interactive shell + set the proper env
+    // vars needed for an interactive shell
+    let cwd = process.cwd();
+    if (options && typeof options.cwd === "string") {
+      cwd = options.cwd;
+    }
+    const cp = spawn(vscode.env.shell, {
+      cwd: cwd,
```
Comment (JohnnyMorganz) [inline editors/code/src/spawn.ts:94]: I think this cwd can be simplified similar to above?
```suggestion
    const cp = spawn(vscode.env.shell, {
      cwd: options?.cwd ?? process.cwd(),
```

File: editors/code/src/spawn.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    spawnNaive(command, args, options)
+      .then(resolve)
+      .catch((err) => {
+        if (!err.ran && canSpawnInteractive(process.platform)) {
+          spawnInteractive(command, args, options).then(resolve).catch(reject);
+        } else {
+          reject(err);
+        }
+      });
+  });
```
Comment (JohnnyMorganz) [inline editors/code/src/spawn.ts:130]: I don't think this needs to be wrapped inside another promise
```suggestion
  return spawnNaive(command, args, options)
      .catch((err) => {
        if (!err.ran && canSpawnInteractive(process.platform)) {
          return spawnInteractive(command, args, options);
        } else {
          return Promise.reject(err);
        }
      });
```

File: editors/code/src/spawn.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+const wrappedSpawn = (
+  command: string,
+  args: string[],
+  options?: SpawnOptionsWithoutStdio
+): Promise<string> => {
+  return new Promise((resolve, reject) => {
+    spawnNaive(command, args, options)
+      .then(resolve)
+      .catch((err) => {
+        if (!err.ran && canSpawnInteractive(process.platform)) {
```
Comment (JohnnyMorganz) [inline editors/code/src/spawn.ts:124]: Where does `err.ran` come from? I can't seem to find it

**Review by JohnnyMorganz — CHANGES_REQUESTED** [review]

File: editors/code/src/spawn.ts (hunk 1 above)
Comment (filiptibell) [reply editors/code/src/spawn.ts:70]: `cwd` should be overwritten by any options passed in with the `...options` on line `72`

File: editors/code/src/spawn.ts (hunk 2 above)
Comment (filiptibell) [reply editors/code/src/spawn.ts:94]: For this one I pulled out `cwd` to be able to pass it as the `PWD` env var on line `97`, is there a more elegant way to do that?

File: editors/code/src/spawn.ts (hunk 4 above)
Comment (filiptibell) [reply editors/code/src/spawn.ts]: Leftover from some earlier testing, removed it!

File: editors/code/src/spawn.ts (hunk 1 above)
Comment (JohnnyMorganz) [reply editors/code/src/spawn.ts:70]: Ah yeah, makes sense, my bad

File: editors/code/src/spawn.ts (hunk 2 above)
Comment (JohnnyMorganz) [reply editors/code/src/spawn.ts:94]: Whoops, didn't see that. I don't think so

**Review by JohnnyMorganz — APPROVED** [review]
Thanks!

**Pull request merged**